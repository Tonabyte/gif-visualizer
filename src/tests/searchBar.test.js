import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

// Components
import SearchBar from '../components/searchBar';

describe('Searchbar Test Suite', () => {
  const props = {
    addCategory: jest.fn(),
  };
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<SearchBar { ...props } />);
  });

  it('Should render Searchbar', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should change text', () => {
    wrapper.find('input').simulate('change', {
      target: {
        value: 'test',
      },
    });
  });

  it('Should submit text', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect(props.addCategory).not.toHaveBeenCalled();
  });

  it('Should clear input text', () => {
    wrapper.find('input').simulate('change', {
      target: {
        value: 'test',
      },
    });
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect(props.addCategory).toHaveBeenCalled();
    expect(wrapper.find('input').prop('value')).toEqual('');
  });
});