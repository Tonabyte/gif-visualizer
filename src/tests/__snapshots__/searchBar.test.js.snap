// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar Test Suite Should render Searchbar 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="¿Qué te gustaría ver?"
    type="text"
    value=""
  />
</form>
`;
